velox/connectors/hive/PartitionIdGenerator.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/PartitionIdGenerator.h"
+
+#include "velox/common/base/Portability.h"
+
+namespace facebook::velox::connector::hive {
+
+PartitionIdGenerator::PartitionIdGenerator(
+    const RowTypePtr& inputType,
+    std::vector<column_index_t> partitionChannels,
+    memory::MemoryPool* FOLLY_NONNULL pool)
+    : partitionChannels_(std::move(partitionChannels)),
+      allPartitionsVector_(std::static_pointer_cast<RowVector>(
+          BaseVector::create(inputType, 0, pool))),
+      activeIds_(SelectivityVector::empty(0)),
+      maxId_(0) {
```

Comment:
nit: you can initialize this member variable when declaring it:

```
uint64_t maxId_{0};
```

Reply:
Revised.